src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -8,69 +8,99 @@ import * as telemetry from '../../shared/telemetry/telemetry'
-    let result: telemetry.Result = 'Succeeded'
-    const response = await new SearchLogGroupWizard().run()
-    if (response) {
-        const logGroupInfo: CloudWatchLogsGroupInfo = {
-            groupName: response.submenuResponse.data,
-            regionName: response.submenuResponse.region,
-        }
+function handleWizardResponse(response: SearchLogGroupWizardResponse, registry: LogStreamRegistry): CloudWatchLogsData {
+    const logGroupInfo = {
+        groupName: response.submenuResponse.data,
+        regionName: response.submenuResponse.region,
+    }
+
+    const parameters: CloudWatchLogsParameters = {
+        limit: registry.configuration.get('limit', 10000),
+        filterPattern: response.filterPattern,
+    }
 
-        const parameters: CloudWatchLogsParameters = {
-            limit: registry.configuration.get('limit', 10000),
-            filterPattern: response.filterPattern,
+    const initialStreamData: CloudWatchLogsData = {
+        data: [],
+        parameters: parameters,
+        busy: false,
+        logGroupInfo: logGroupInfo,
+        retrieveLogsFunction: filterLogEventsFromUriComponents,
+    }
+    return initialStreamData
+}
+
+async function prepareDocument(uri: vscode.Uri, registry: LogStreamRegistry): Promise<void> {
+    const doc = await vscode.workspace.openTextDocument(uri) // calls back into the provider
+    vscode.languages.setTextDocumentLanguage(doc, 'log')
+
+    const textEditor = await vscode.window.showTextDocument(doc, { preview: false })
+    registry.setTextEditor(uri, textEditor)
+
+    // Initial highlighting of the document and then for any addLogEvent calls.
+    highlightDocument(registry, uri)
+    vscode.workspace.onDidChangeTextDocument((event: vscode.TextDocumentChangeEvent) => {
+        if (event.document.uri.toString() === doc.uri.toString()) {
+            highlightDocument(registry, uri)
         }
+    })
+}
 
-        const uri = createURIFromArgs(logGroupInfo, parameters)
-        const initialStreamData: CloudWatchLogsData = {
-            data: [],
-            parameters: parameters,
-            busy: false,
-            logGroupInfo: logGroupInfo,
-            retrieveLogsFunction: filterLogEventsFromUriComponents,
+export async function searchLogGroup(node: LogGroupNode | undefined, registry: LogStreamRegistry): Promise<void> {
+    let result: telemetry.Result = 'Succeeded'
+    let response: SearchLogGroupWizardResponse | undefined
+
+    if (node) {
+        if (!node.logGroup.logGroupName) {
+            throw new Error('CWL: Log Group node does not have a name.')
         }
         // Currently displays nothing if update log fails in non-cancellationError. (don't want this)
 
+        response = await new SearchLogGroupWizard({
+            groupName: node.logGroup.logGroupName,
+            regionName: node.regionCode,
+        }).run()
+    } else {
+        response = await new SearchLogGroupWizard().run()
+    }
+
+    if (response) {
```

Comment:
Better to return early to ease the nesting:
```suggestion
    if (!response) {
        telemetry.recordCloudwatchlogsOpenStream({ result: 'Cancelled' })
        return
    }
```